src/core/net/netif.hpp

```diff
@@ -156,6 +157,47 @@ class NetifMulticastAddress : public otNetifMulticastAddress,
     bool Matches(const Address &aAddress) const { return GetAddress() == aAddress; }
 };
 
+class ExternalNetifMulticastAddress : public NetifMulticastAddress
+{
+    friend class Netif;
+    friend class LinkedList<ExternalNetifMulticastAddress>;
```

Comment:
`LinkedList<ExternalNetifMulticastAddress>` is necessary not because we have an extra list of `ExternalNetifMulticastAddress`, but because of the fact that **Pool<ExternalNetifMulticastAddress> uses a free list of `ExternalNetifMulticastAddress` for allocation**.

Reply:
I see. Makes sense. Thanks.

Is that why we also need the `GetNext()` implementation (to cast the `mNext` to `ExtenralMultiAddr` in this class?

Reply:
Yes. But I defined `GetNext` in public so `friend class` is actually not necessary. 
Agree that it's better to  `GetNext()` to private.